sw/device/tests/chip_power_sleep_load.c

```diff
@@ -0,0 +1,377 @@
+
+  expected_isr_handler();
+
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+  CHECK_DIF_OK(dif_rv_core_ibex_clear_nmi_state(&rv_core_ibex,
+                                                kDifRvCoreIbexNmiSourceAll));
+}
+
+static void wdog_irq_handler(void) {
+  bool is_pending;
+
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      &aon_timer, kDifAonTimerIrqWdogTimerBark, &is_pending));
+  irq_is_pending = is_pending;
+
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon_timer));
+
+  CHECK_DIF_OK(
+      dif_aon_timer_irq_acknowledge(&aon_timer, kDifAonTimerIrqWdogTimerBark));
+}
+
+void prepare_to_exit(void) {
+  CHECK_DIF_OK(dif_pwm_phase_cntr_set_enabled(&pwm, kDifToggleDisabled));
+
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 0));
+
+  LOG_INFO("SW_INFO: Prepare to exit");
+
+  CHECK(ext_irq_fired == false, "Unexpected external interrupt triggered.");
+
+  pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0);
+
+  aon_timer_testutils_shutdown(&aon_timer);
+
+  test_status_set(kTestStatusInTest);
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rv_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR), &rv_timer));
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+  CHECK_DIF_OK(dif_pwm_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWM_AON_BASE_ADDR), &pwm));
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp_ctrl));
+  CHECK_DIF_OK(
+      dif_gpio_init(mmio_region_from_addr(TOP_EARLGREY_GPIO_BASE_ADDR), &gpio));
+
+  LOG_INFO("SW_INFO: Running CHIP Power Sleep Load test");
+
+  static const uint32_t kGpioMask = 0x00000004;
+
+  CHECK_DIF_OK(
+      dif_pinmux_output_select(&pinmux, (kTopEarlgreyPinmuxMioOutIoa0 + 2),
+                               (kTopEarlgreyPinmuxOutselGpioGpio0 + 2)));
+
+  // Set output modes of all GPIO pins
+  CHECK_DIF_OK(dif_gpio_output_set_enabled_all(&gpio, kGpioMask));
+
+  // Write to set IOA2 low at the start of the test:
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 0));
+
+  LOG_INFO("SW_INFO: GPIO active");
+
+  // PWRMGR- Read wakeup reason
+
+  dif_pwrmgr_wakeup_reason_t wakeup_reason;
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_get(&pwrmgr, &wakeup_reason));
+  LOG_INFO("SW_INFO: wakeup type:%d. wakeup reason: 0x%02X",
+           wakeup_reason.types, wakeup_reason.request_sources);
+
+  if (wakeup_reason.types == 0) {
```

Comment:
is this supposed to be an empty block? can you remove this and change the `else` below to an `else if` ?

Reply:
Yes, this is supposed to be empty.
In case of PORESET, I want the program to continue, i.e.: not finish the test with true or print error and exit the test with false.
This seemed easier and more readable than putting the rest of the test in brackets or making it into a function.
I added a comment for this.